File: app/src/main/java/com/example/kotlincashloan/ui/notification/GcmBroadcastReceiver.java
package com.example.kotlincashloan.ui.notification;

import android.app.Activity;
import android.content.BroadcastReceiver;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;

import androidx.appcompat.app.AppCompatActivity;
import androidx.legacy.content.WakefulBroadcastReceiver;

import com.timelysoft.tsjdomcom.service.AppPreferences;

public class GcmBroadcastReceiver extends WakefulBroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        // Explicitly specify that GcmIntentService will handle the intent.
        ComponentName comp = new ComponentName(context.getPackageName(), PushNotification.class.getName());
        // Start the service, keeping the device awake while it is launching.=
        startWakefulService(context, (intent.setComponent(comp)));
        setResultCode(Activity.RESULT_OK);
    }
}

File: app/src/main/java/com/example/kotlincashloan/ui/notification/GcmBroadcastReceiver.kt
package com.example.kotlincashloan.ui.notification

import android.app.*
import android.content.*
import android.os.Bundle
import android.util.Log
import androidx.legacy.content.WakefulBroadcastReceiver.startWakefulService
import com.timelysoft.tsjdomcom.service.AppPreferences.dataKey
import com.timelysoft.tsjdomcom.service.AppPreferences.init
import java.util.*


class GcmBroadcastReceiver : BroadcastReceiver() {

    var ctx: Context? = null
    override fun onReceive(context: Context, intent: Intent) {
        init(context)
        // Explicitly specify that GcmIntentService will handle the intent.
        val i = Intent(".ui.notification.PushNotification")
        i.setClass(context, PushNotification::class.java)
        context.startService(i)

        val comp = ComponentName(context.packageName, PushNotification::class.java.name)
        // Start the service, keeping the device awake while it is launching.=
        ctx = context
        postNotification(Objects.requireNonNull(intent.extras)!!)
        startWakefulService(context, intent.setComponent(comp))
        resultCode = Activity.RESULT_OK
    }

    // post GCM message to notification center.
    private fun postNotification(data: Bundle) {
        val msg = data.getString("alert")
        Log.i(ContentValues.TAG, "message: $msg")
        if (msg != null) // on app startup, this was always getting called with empty message
            return
        dataKey = data.getString("action", "")
    }
}